rest/route/subscription_routes.go

```diff
@@ -154,3 +159,59 @@ func (s *subscriptionGetHandler) Execute(_ context.Context, sc data.Connector) (
 		Result: model,
 	}, nil
 }
+
+type subscriptionDeleteHandler struct {
+	id bson.ObjectId
+}
+
+func (s *subscriptionDeleteHandler) Handler() RequestHandler {
+	return &subscriptionDeleteHandler{}
+}
+
+func (s *subscriptionDeleteHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	u := MustHaveUser(ctx)
+	idString := r.FormValue("id")
+	if idString == "" {
+		return rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "Must specify an ID to delete",
+		}
+	}
+	if !bson.IsObjectIdHex(idString) {
+		return rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "Invalid ID format",
+		}
+	}
+	s.id = bson.ObjectIdHex(idString)
+	subscription, err := event.FindSubscriptionByID(s.id)
+	if err != nil {
+		return rest.APIError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    "Problem finding subscription",
+		}
+	}
+	if subscription == nil {
+		return rest.APIError{
+			StatusCode: http.StatusNotFound,
+			Message:    "Subscription not found",
+		}
+	}
+	if subscription.Owner != u.Username() {
+		return rest.APIError{
+			StatusCode: http.StatusUnauthorized,
```

Comment:
Should we log this at, say, Critical, since this should _never_ happen?

Reply:
It shouldn't ever happen, but I don't really see why this specifically should log when there are other more useful avenues for attack

Reply:
I didn't mean in the sense of an attack, but in the sense that if this happens, there was likely programmer error of some kind. I still think we should log this at some level.

Reply:
It definitely shouldn't happen, but I feel weird logging this when there are many other things that shouldn't happen but aren't logged